sql/telemetry_derived/italy_covid19_outage_v1/query.sql

```diff
@@ -0,0 +1,284 @@
+    WHERE
+      LENGTH(entry) > 0
+  )
+);
+
+-- Get a stable source for DAUs.
+WITH DAUs AS (
+  SELECT
+    submission_date AS date,
+    COUNT(*) AS client_count
+  FROM
+    telemetry.clients_daily
+  WHERE
+    submission_date >= '2020-01-01'
+    AND submission_date <= '2020-03-31'
+    AND country = 'IT'
+  GROUP BY
+    1
+  HAVING
+    -- The minimum size of a bucket for it to be published.
+    COUNT(*) > 5000
+),
+-- Compute aggregates for the health data.
+health_data_sample AS (
+  SELECT
+    DATE(SAFE_CAST(creation_date AS TIMESTAMP)) AS date,
+    client_id,
+    SUM(
+      coalesce(
+        CAST(JSON_EXTRACT(additional_properties, '$.payload.sendFailure.undefined') AS INT64),
+        0
+      )
+    ) AS eUndefined,
+    SUM(
+      coalesce(
+        CAST(JSON_EXTRACT(additional_properties, '$.payload.sendFailure.timeout') AS INT64),
+        0
+      )
+    ) AS eTimeOut,
+    SUM(
+      coalesce(CAST(JSON_EXTRACT(additional_properties, '$.payload.sendFailure.abort') AS INT64), 0)
+    ) AS eAbort,
+    SUM(
+      coalesce(
+        CAST(JSON_EXTRACT(additional_properties, '$.payload.sendFailure.eUnreachable') AS INT64),
+        0
+      )
+    ) AS eUnreachable,
+    SUM(
+      coalesce(
+        CAST(JSON_EXTRACT(additional_properties, '$.payload.sendFailure.eTerminated') AS INT64),
+        0
+      )
+    ) AS eTerminated,
+    SUM(
+      coalesce(
+        CAST(JSON_EXTRACT(additional_properties, '$.payload.sendFailure.eChannelOpen') AS INT64),
+        0
+      )
+    ) AS eChannelOpen,
+  FROM
+    telemetry.health
+  WHERE
+    date(submission_timestamp) >= '2020-01-01'
+    AND date(submission_timestamp) <= '2020-03-31'
+    AND normalized_country_code = 'IT'
+  GROUP BY
+    1,
+    2
+),
+health_data_aggregates AS (
+  SELECT
+    date,
+    COUNTIF(eUndefined > 0) AS num_clients_eUndefined,
+    COUNTIF(eTimeOut > 0) AS num_clients_eTimeOut,
+    COUNTIF(eAbort > 0) AS num_clients_eAbort,
+    COUNTIF(eUnreachable > 0) AS num_clients_eUnreachable,
+    COUNTIF(eTerminated > 0) AS num_clients_eTerminated,
+    COUNTIF(eChannelOpen > 0) AS num_clients_eChannelOpen,
+  FROM
+    health_data_sample
+  GROUP BY
+    date
+  HAVING
+    COUNT(*) > 5000
+),
+final_health_data AS (
+  SELECT
+    h.date,
+    (num_clients_eUndefined / DAUs.client_count) AS proportion_undefined,
+    (num_clients_eTimeOut / DAUs.client_count) AS proportion_timeout,
+    (num_clients_eAbort / DAUs.client_count) AS proportion_abort,
+    (num_clients_eUnreachable / DAUs.client_count) AS proportion_unreachable,
+    (num_clients_eTerminated / DAUs.client_count) AS proportion_terminated,
+    (num_clients_eChannelOpen / DAUs.client_count) AS proportion_channel_open,
+  FROM
+    health_data_aggregates AS h
+  INNER JOIN
+    DAUs
+  ON
+    DAUs.date = h.date
+),
+-- Compute aggregates for histograms coming from the health ping.
+histogram_data_sample AS (
+  SELECT
+    client_id,
+    document_id,
```

Comment:
This field appears to be unused, please remove it

Reply:
Cheers, thanks for spotting! Done!